src/main/java/calculator/service/StringCalculate.java

```diff
@@ -0,0 +1,80 @@
+package calculator.service;
+
+import calculator.constant.Constant;
+import calculator.model.Calculator;
+import calculator.validator.InputValidator;
+
+import java.util.ArrayList;
+
+public class StringCalculate {
+
+    private static final String REGEX_START = "\\";
+    private static final String REGEX_OR = "|";
+
+    private String inputString;
+    Calculator calculator = Calculator.getInstance();
+
+    public StringCalculate(String inputString) {
+        this.inputString = inputString;
+    }
+
+    public void preprocessDelimiter() {
+        findCustomDelimiter();
+        takeOffCustomDelimiter();
+    }
+
+    private void findCustomDelimiter() {
+        if (inputString.startsWith(Constant.CUSTOM_DELIMITER_PREFIX)) {
+            char newDelimiter = inputString.charAt(2);
+            calculator.addCustomDelimiter(newDelimiter);
+        }
+    }
+
+    private void takeOffCustomDelimiter() {
+        if (inputString.startsWith(Constant.CUSTOM_DELIMITER_PREFIX)) {
+            inputString = inputString.substring(5);
+        }
+    }
+
+    public void splitAndSaveOperands() {
+        String[] stringOperands = splitOperands();
+        InputValidator.validateStringOperand(stringOperands);
+        ArrayList<Integer> intOperands = convertOperandsToInt(stringOperands);
+        saveOperands(intOperands);
+    }
+
+    private String[] splitOperands() {
+        String reg = "";
+        for (Character delimiter : calculator.getDelimiters()) {
+            reg += (REGEX_START + delimiter + REGEX_OR);
+        }
+        if (!reg.isEmpty()) {
+            reg = reg.substring(0, reg.length() - 1);
+        }
+        return inputString.split(reg, -1);
+    }
+
+    private ArrayList<Integer> convertOperandsToInt(String[] stringOperands) {
+        ArrayList<Integer> intOperands = new ArrayList<>();
+        for (String stringOperand : stringOperands) {
+            if (stringOperand.equals(Constant.BLANK)) {
+                intOperands.add(Constant.ZERO);
+            } else {
+                intOperands.add(Integer.valueOf(stringOperand));
+            }
```

Comment:
개인 취향일 수도 있으나, 블록의 depth를 짧게하는 것을 좋아하기 때문에 해당 부분을 삼항 연산자로 풀어도 좋을 것 같습니다.

intOperands.add 는 동일한 기능이고 들어가는 인자가 다르다면 해당 부분에 대한 Value를 뽑아주는 메소드를 생성하면 좋을 것 같습니다.

```java
private Integer convertToIntegerOperand(String stringOperand){
    return stringOperand.eqauls(Constant.BLANK) ? Constant.ZERO : Integer.valueOf(StringOperand);
}
```
그래서, 원래 부분을 다음과 같이 한줄로 극복 가능합니다.
```suggestion
                            intOperands.add(convertToIntegerOperand(stringOperand));
```

Reply:
훨씬 보기에도 편하고 깔끔한 방법이네요! 좋은 지적 감사합니다👍